tests/integrationv2/providers.py

```diff
@@ -308,8 +311,95 @@ def setup_server(self):
         return cmd_line
 
 
-class OpenSSL(Provider):
+class CriterionS2N(S2N):
+    """
+    Wrap the S2N provider in criterion-rust
+    """
+    criterion_off = 'off'
+    criterion_delta = 'delta'
+    criterion_baseline = 'baseline'
+    # Figure out what mode to run in: baseline or delta
+    criterion_mode = get_flag(S2N_USE_CRITERION)
+
+    def _find_s2n_benchmark(self, pattern):
+        # Use executable bit to find the correct file.
+        result = find_files(pattern, root_dir=self.cargo_root, modes=['0o775','0o755'])
+        if len(result) != 1:
+            raise FileNotFoundError(
+                f"Exactly one s2n criterion benchmark not found {result} with expected file mask.")
+        else:
+            return result[0]
+
+    def _find_cargo(self):
+        # return a path to the highest level dir containing Cargo.toml
+        shortest = None
+        for file in find_files("Cargo.toml", root_dir="../.."):
+            if shortest is None:
+                shortest = file
+            else:
+                if len(file) < len(shortest):
+                    shortest = file
+        if shortest is None:
+            raise FileNotFoundError("Unable to find Cargo.toml")
+        # Return the path, minus Cargo.toml
+        return os.path.dirname(shortest)
+
+    def _cargo_bench(self):
+        """
+        Find the handlers
+        """
+        self.s2nc_bench = self._find_s2n_benchmark("s2nc-")
+        self.s2nd_bench = self._find_s2n_benchmark("s2nd-")
+
+    def __init__(self, options: ProviderOptions):
+        super().__init__(options)
+        # Set cargo root
+        self.cargo_root = self._find_cargo()
+
+        # Find the criterion binaries
+        self._cargo_bench()
+
+        # strip off the s2nc/d at the front because criterion
+        if 's2nc' in self.cmd_line[0] or 's2nd' in self.cmd_line[0]:
+            self.cmd_line = self.cmd_line[1:]
+
+        # strip off the s2nc -e argument, criterion handler isn't sending any STDIN characters,
+        # and makes it look like s2nc is hanging.
+        if '-e' in self.cmd_line:
+            self.cmd_line.remove('-e')
+            print(f"***** cmd_line is now {self.cmd_line}")
 
+        # Copy the command arguments to an environment variable for the harness to read.
```

Comment:
Why can't the harness accept the arguments as arguments? Is the environment variable necessary?

Reply:
Was unable to find a way to access the arguments in the handler, the env. variable seemed the cleanest at the time.